doc/advanced_data_structures.rst

```diff
@@ -0,0 +1,117 @@
+******************************
+Advanced Data Structure Topics
+******************************
+
+This page of documentation is reserved for advanced topics in SmartRedis
+data structures that may not be needed for all users.
+
+.. _advanced_data_structures_dataset_aggregation:
+
+Dataset Aggregation
+===================
+
+In addition to the ability to work with individual datasets, SmartRedis offers
+users the manage lists of datasets and work with them collectively. This is
+particularly useful for producer/consumer relationships.
+
+The DataSet Aggregation API stores multiple ``DataSet`` objects
+on one or more database nodes through an interface referred to as
+``aggregation lists``.
+An ``aggregation list`` in SmartRedis stores references to
+``DataSet`` objects that are stored in the database.  The SmartRedis client
+in one application can append ``DataSet`` objects to the ``aggregation list``;
+subsequently, ``SmartRedis`` clients in the same application or a different
+application can retrieve some or all of the ``DataSet`` objects referenced
+in that ``aggregation list``.
+
+Appending to a DataSet
+----------------------
+
+For example, the C++ client function to append a ``DataSet`` to an
+aggregation list is shown below:
+
+.. code-block:: cpp
+
+    # C++ aggregation list append interface
+    void append_to_list(const std::string& list_name,
+                        const DataSet& dataset);
+
+The above function will append the provided ``DataSet`` to the
+``aggregation list``, which can be referenced in all user-facing functions
+by the provided list name.  Note that a list can be appended by
+any client in the same or different application.  Implicitly, all
+DataSets, when appended, are added to the end of the list. If the list does not
+already exist, it is automatically created.
+
+Retrieving DataSets from an aggregation list
+--------------------------------------------
+
+To retrieve ``aggregation list`` contents,
```

Comment:
To retrieve the contents of an ``aggregation list``

Reply:
Fixed